packages/flutter/test/widgets/framework_test.dart

```diff
@@ -451,6 +451,41 @@ void main() {
     expect(count, 2);
   });
 
+  testWidgets('GlobalKey - detach and re-atach child to different parents', (WidgetTester tester) async {
```

Comment:
the 'Skipping tabs with global key does not crash' test in flutter/packages/flutter/test/material/tabs_test.dart will hit this edge case

This only becomes an issue when we implement move. Combining with the performLayout rounding error. It is a bunch of coincident that cause it.
I wrote this test just to simplify situation, so it will be easier to understand what is going on.

I do think this might happen sometime in the future if developer start to write complex program.

If we want to keep this as is. We will have to fix the rounding issue in perform layout